chore: sync database copy with seed contents

MikeMcC399 opened the pull request:
This PR syncs [data/database.json](https://github.com/cypress-io/cypress-realworld-app/blob/develop/data/database.json) from the reference copy in [data/database-seed.json](https://github.com/cypress-io/cypress-realworld-app/blob/develop/data/database-seed.json).

This resolves the issue that running `yarn dev` changes `data/database.json` and since this database file is committed to `git` it causes a `git` difference to be reported. It is not normally expected that testing would change the contents of any committed files. [README > Run the app](https://github.com/cypress-io/cypress-realworld-app#run-the-app) specifies `yarn dev` as the correct command to run the app in preparation for testing. It uses `yarn predev` to copy the database seed into the working database.

## Verification

Check `git status` before and after executing `yarn dev` and confirm that there is no change.

Test using `yarn cypress:run` and `yarn cypress:run:component`

MikeMcC399 committed:
fix: repopulate database.json from seed

MikeMcC399 commented:
- CI fails due to flake in open issue https://github.com/cypress-io/cypress-realworld-app/issues/1332

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/cypress-io/cypress-realworld-app?pullRequest=1345) <br/>All committers have signed the CLA.

MikeMcC399 commented:
Force pushed to get around unsigned CLA from @marktnoonan and to trigger a re-test.

marktnoonan commented:
Oh funny @MikeMcC399 looks like I triggered a CLA recheck anyway that worked. I'm working on getting a couple reviewers from folks who own this repo. It seems fine to me but I haven't taken time yet to get familiar with this project in a lot of detail.

MikeMcC399 commented:
@marktnoonan 

> Oh funny @MikeMcC399 looks like I triggered a CLA recheck anyway that worked. I'm working on getting a couple reviewers from folks who own this repo. It seems fine to me but I haven't taken time yet to get familiar with this project in a lot of detail.

Always good to get additional eyes on any changes! I have only scratched the surface of understanding this repo myself.

When you merged master into this branch it showed you as author and apparently you have never contributed to this repo, so it wanted a CLA commitment from you. When I rebased, that overwrote your change and so you were no longer a co-author and CLA wasn't interested in you anymore!

I rebased primarily because there was an unexplained CI error which seemed unrelated to this change. That is the only way I can retrigger tests in CircleCI.

There is a mixture of flakiness in these tests.

elylucas commented:
Hi @MikeMcC399,

We are going to do a review and updates to this repo similar to the RWT updates last week, so if you have any feedback let us know! This repo is more engrained into the Cypress ecosystem overall, so changes probably won't be as quick as the last one.

MikeMcC399 commented:
@elylucas 

Thank you for asking! Apart from the open issues that would need to be addressed, I would suggest:

1. migrate to a later supported version of Node.js since `16.x` enters end-of-life in Sep 2023.
2. replace CRA (perhaps use Next.js?). The last release of CRA was more than one year ago.
